### PR TITLE
Geo/partial saturation tests

### PR DESCRIPTION
**Pull request by WPK4FEM**
**📝 Description**
Changes to partially saturated tests.

**🆕 Changelog**
- Changed the tests using steady state Pw to transient Pw ( I think that was originally the intention )
- Added suction ( positive pressure ) on the hydrostatic boundary conditions, to get away from the step function for the relative permeability, Bishop coefficient etc ) at zero pressure
- added test ( for regression, no documentation so far ) for climbing and falling phreatic line during a time span.

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py [hunk 1]
```diff
@@ -44,14 +44,14 @@ def __test_saturated_below_phreatic_level_pw(self, test_name):
...
-        rel_p_stage  = [self.__compute_hydrostatic_water_pressure(y_coord) for y_coord in y_coords]
+        rel_p_stage  = [self.__compute_hydrostatic_water_pressure(y_coord, -2.0) for y_coord in y_coords]
         
         errors_stage = [actual_pressure - expected_pressure for actual_pressure, expected_pressure in
                         zip(stage_water_pressure[1], rel_p_stage)]
         rmse_stages = (sum([error ** 2 for error in errors_stage]) / len(errors_stage)) ** 0.5
 
         # assert if average error in all stages is below accuracy
-        accuracy = 1.0e-7
+        accuracy = 1.0e-3
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partial_saturation.py:54]: The accuracy is decreased here. It may happen that the tests are still not converging, but it hits thir criterium?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 2]
```diff
@@ -0,0 +1,179 @@
+{
+  "problem_data": {
+    "problem_name": "mesh",
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: Problem_name preferly to describe the type of problem. Setting it as "mesh" is not descriptive.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    "start_time": 0.0,
+    "end_time": 29.0,
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:5]: Is there any specific reason to set it to 29?  Why not 30?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    "model_import_settings": {
+      "input_type": "mdpa",
+      "input_filename": "mesh"
+    },
+    "material_import_settings": {
+      "materials_filename": "MaterialParameters.json"
+    },
+    "time_stepping": {
+      "time_step": 1.0,
+      "max_delta_time_factor": 1.0
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: As increase_factor is set to 1, this "max_delta_time_factor" is not necessary.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+    "buffer_size": 2,
+    "echo_level": 1,
+    "clear_storage": false,
+    "compute_reactions": true,
+    "reform_dofs_at_each_step": false,
+    "nodal_smoothing": false,
+    "block_builder": true,
+    "solution_type": "Quasi-Static",
+    "scheme_type": "Backward_Euler",
+    "reset_displacements": true,
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: This is a single stage calculation. Do we still need reset_displacement?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    "strategy_type": "newton_raphson",
+    "convergence_criterion": "residual_criterion",
+    "residual_relative_tolerance": 0.001,
+    "residual_absolute_tolerance": 1E-09,
+    "min_iterations": 6,
+    "max_iterations": 10,
+    "number_cycles": 1,
+    "reduction_factor": 1.0,
+    "increase_factor": 1.0,
+    "calculate_reactions": true,
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:42]: I see "compute_reactions": true  above. Here "calculate_reactions": true. I am just curious what is the difference between them?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    "processes_sub_model_part_list": [
+      "gravity",
+      "TopWaterPressure",
+      "BottomWaterPressure",
+      "InitialWaterPressure"
+    ],
+    "body_domain_sub_model_part_list": [
+      "Soil"
+    ],
+    "move_mesh_flag": false
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: Preferly shift "move_mesh_flag": false to above, before "linear_solver_settings". To be consistent with the other variables

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+              "plane_output": [],
+              "nodal_results": [
+                "WATER_PRESSURE",
+                "DT_WATER_PRESSURE"
+              ],
+              "gauss_point_results": [
+                "FLUID_FLUX_VECTOR",
+                "HYDRAULIC_HEAD",
+                "DEGREE_OF_SATURATION",
+                "BISHOP_COEFFICIENT"
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:94]: "BISHOP_COEFFICIENT" is given as constanr by the user. Does it change during partially saturated calculation? If not, then not necessary to give it as output.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    "constraints_process_list": [
+      {
+        "python_module": "apply_scalar_constraint_table_process",
+        "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name": "ApplyScalarConstraintTableProcess",
+        "Parameters": {
+          "model_part_name": "PorousDomain.TopWaterPressure",
+          "variable_name": "WATER_PRESSURE",
+          "is_fixed": true,
+          "table": [1,1],
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:112]: A bit confused here. The Pressure is scalar, but "table": [1,1] is used, while I am expecting "table":1. Or maybe "Phreatic_Multi_Line" requires this?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+      },
+      {
+        "python_module": "apply_scalar_constraint_table_process",
+        "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name": "ApplyScalarConstraintTableProcess",
+        "Parameters": {
+          "model_part_name": "PorousDomain.BottomWaterPressure",
+          "variable_name": "WATER_PRESSURE",
+          "is_fixed": true,
+          "table": [1,1],
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:131]: Same as above, "table": 1?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+      },
+      {
+        "python_module": "apply_scalar_constraint_table_process",
+        "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name": "ApplyScalarConstraintTableProcess",
+        "Parameters": {
+          "model_part_name": "PorousDomain.InitialWaterPressure",
+          "variable_name": "WATER_PRESSURE",
+          "is_fixed": false,
+          "table": [0,0],
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:150]: Same as above

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/mesh.mdpa [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+  End SubModelPartTables
+  Begin SubModelPartNodes
+    2
+    3
+  End SubModelPartNodes
+End SubModelPart
+
+Begin SubModelPart BottomWaterPressure
+  Begin SubModelPartTables
+    1
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/mesh.mdpa:94]: The TopWaterPressure and BottomWaterPressure share the same table. Is it done intentially?

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_saturated_below_phreatic_level_pw_triangle3N/ProjectParameters_stage2.json [hunk 13]
```diff
@@ -19,7 +19,7 @@
       "materials_filename"             : "../Common/MaterialParameters_stage2.json"
     },
     "time_stepping": {
-      "time_step"                      : 1.0,
+      "time_step"                      : 0.1,
       "max_delta_time_factor"          : 1.0
```
Comment (mnabideltares) [inline applications/GeoMechanicsApplication/tests/test_partially_saturated/test_saturated_below_phreatic_level_pw_triangle3N/ProjectParameters_stage2.json]: "increase_factor" : 1.0,  then "max_delta_time_factor"  not necessary

**Review by mnabideltares — COMMENTED** [review]
@WPK4FEM 
Thank you for this nice work.
Generally looks good to me, however I have minor comments.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 2 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: Changed to "rising_falling_phreatic_level"

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 3 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:5]: It is designed this way to have regular human recognizable phreatic levels a the chosen time points.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 4 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: removed.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 5 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: No, strictly speaking not. Removed the setting.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 6 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:42]: They are both settings for core and have to do with computing the unbalance and computing nodal reaction forces for output. I will try to find out which does what outside the scope of this PR.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 7 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json]: changed accordingly

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 8 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:94]: BISHOP_COEFFICIENT is not set by the user, it is a result of the computation. For the saturated_below_phreatic_line retention law it is 0 in dry zone above the phreatic surface, 1 in the wet zone below phreatic surface.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 9 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:112]: There are 2 points that define the Phreatic_Multi_line  here. The tables give the change in head for each point in time.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 10 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:131]: see previous response.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json (hunk 11 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/ProjectParameters.json:150]: see previous response

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/mesh.mdpa (hunk 12 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_rising_falling_phreatic_level_pw_quad4N/mesh.mdpa:94]: Yes this is intentional. Boundary conditions are chosen such that the same phreatic level is set on both ends of the mesh.

File: applications/GeoMechanicsApplication/tests/test_partially_saturated/test_saturated_below_phreatic_level_pw_triangle3N/ProjectParameters_stage2.json (hunk 13 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partially_saturated/test_saturated_below_phreatic_level_pw_triangle3N/ProjectParameters_stage2.json]: removed the setting here and all other project parameters files within test_partial_saturation

File: applications/GeoMechanicsApplication/tests/test_partial_saturation.py (hunk 1 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/test_partial_saturation.py:54]: Checked that they are converging.

**Review by mnabideltares — APPROVED** [review]
@WPK4FEM 
Thank you for changes. It looks OK to me
